client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -20,9 +21,13 @@ private final class PoolManager[F[_], A <: Connection[F]](
     responseHeaderTimeout: Duration,
     requestTimeout: Duration,
     semaphore: Semaphore[F],
-    implicit private val executionContext: ExecutionContext)(implicit F: Concurrent[F])
+    implicit private val executionContext: ExecutionContext,
+    statsPublisher: Option[PoolManagerStats => Unit],
```

Comment:
How about `PoolManagerStats => IO[Unit]`, with a default implementation of `_ => IO.unit`?  Just to avoid `Concurrent` overhead on something that's `F.unit`?

Reply:
That would be nice, although I believe it would need an implict `ContextShfit[IO]` in order to schedule the work on a separate fiber (if using a fiber is a valid approach here!). We'd also then need to lift the result of starting the fiber into `F` anyway.